Source/JavaScriptCore/bytecompiler/BytecodeGenerator.cpp

```diff
@@ -951,9 +951,11 @@ BytecodeGenerator::BytecodeGenerator(VM& vm, EvalNode* evalNode, UnlinkedEvalCod
     m_cachedParentTDZ = parentScopeTDZVariables;
 
     emitEnter();
-
     allocateScope();
-    
+    m_topLevelScopeRegister = addVar();
+    m_topLevelScopeRegister->ref();
+    move(m_topLevelScopeRegister, scopeRegister());
```

Comment:
We hav four `BytecodeGenerator::BytecodeGenerator` constructors. So, why define `m_topLevelScopeRegister` only for these two constructors?

Reply:
This move is wasteful when we do not need this topLevelScopeRegister, and only module and eval need them. Only materializing them on these types is important to save bytecodes in functions.